### PR TITLE
Scale down enhancements

### PR DESCRIPTION
**Pull request by sandoichi**
- Handle scenarios in which we are scaling down and racks are currently unbalanced/set up in weird configurations (2 - 2 - 4 for example)
- Add integration tests for unbalanced rack scale down
- Simplify the naming of the `DatacenterConditionValid` condition

### CODE REVIEW

**Review by respringer — APPROVED** [review]
Looks good. +1

File: README.md [hunk 1]
```diff
@@ -243,6 +243,7 @@ helm install --set image=docker.pkg.github.com/datastax/cass-operator/operator:l
 - Rolling reboot nodes by changing the CRD
 - Store data in a rack-safe way - one replica per cloud AZ
 - Scale up racks evenly with new nodes
+- Scale down racks evenly by decommissioning existing nodes
```
Comment (jimdickinson) [inline README.md:246]: 💯

File: operator/pkg/reconciliation/decommission_node.go [hunk 2]
```diff
@@ -17,13 +17,53 @@ import (
...
 func (rc *ReconciliationContext) DecommissionNodes(epData httphelper.CassMetadataEndpoints) result.ReconcileResult {
 	logger := rc.ReqLogger
 	logger.Info("reconcile_racks::DecommissionNodes")
 	dc := rc.Datacenter
 
-	for idx := range rc.desiredRackInformation {
-		rackInfo := rc.desiredRackInformation[idx]
+	var currentSize int32
+	for _, sts := range rc.statefulSets {
+		currentSize += *sts.Spec.Replicas
```
Comment (jimdickinson) [inline operator/pkg/reconciliation/decommission_node.go:52]: should we nil check here?

File: tests/scale_down_unbalanced_racks/scale_down_unbalanced_racks_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+			step = "check that the extra pod is ready"
+			json = "jsonpath={.items[*].status.containerStatuses[0].ready}"
+			k = kubectl.Get("pod").
+				WithFlag("field-selector", fmt.Sprintf("metadata.name=%s", extraPod)).
+				FormatOutput(json)
+			ns.WaitForOutputAndLog(step, k, "true", 360)
+
+			// The rack with an extra node should get a decommission request
+			// first, despite being the last rack and the first rack also needing
+			// to eventually decommission nodes
```
Comment (jimdickinson) [inline tests/scale_down_unbalanced_racks/scale_down_unbalanced_racks_suite_test.go:85]: 👍

File: docs/user/README.md [hunk 4]
```diff
@@ -437,6 +437,22 @@ Kubernetes worker nodes available.
 For racks to act effectively as a fault-containment zone, each rack in the
 cluster must contain the same number of instances.
 
+## Scale down
```
Comment (jimdickinson) [inline docs/user/README.md:440]: 👍

**Review by jimdickinson — APPROVED** [review]
looks great!

**Pull request merged**